### PR TITLE
handle async check_whitelist in app

### PR DESCRIPTION
**Pull request by minrk**
follow-up to previous PR

### CODE REVIEW

**Comment by minrk** [comment]
follow-up to #876

File: jupyterhub/app.py [hunk 1]
```diff
@@ -980,11 +982,12 @@ def _add_tokens(self, token_dict, kind):
...
+    
+    @gen.coroutine
     def init_api_tokens(self):
         """Load predefined API tokens (for services) into database"""
-        self._add_tokens(self.service_tokens, kind='service')
-        self._add_tokens(self.api_tokens, kind='user')
+        yield self._add_tokens(self.service_tokens, kind='service')
+        yield self._add_tokens(self.api_tokens, kind='user')
     
     def init_services(self):
```
Comment (willingc) [inline jupyterhub/app.py]: @minrk Does this need `@gen.coroutine` too?

Comment (minrk) [inline jupyterhub/app.py]: it doesn't, only those that call async methods need it.

Comment (willingc) [inline jupyterhub/app.py]: Got it. Thanks.

**Pull request merged**